### PR TITLE
Add PyVista (...it's pretty great)

### PR DESCRIPTION
**Pull request by leycec**
This commit adds [**PyVista**](https://docs.pyvista.org), popular quasi-official Python bindings for [**VTK**](https://vtk.org), a popular C++ framework for scientific 3D visualization. Both are maintained by [Kitware](https://www.kitware.com) – an academic open-source company sorta resembling the Qt Company in structure and practice.

Specifically, this commit adds:

* `dev-python/pyvista` itself, of course.
* Three currently unpackaged Python packages required by `dev-python/pyvista`:
  * `dev-python/meshio`, with full unit test support. Of the three dependencies, [meshio](https://pypi.org/project/meshio) is the most interesting; it's a high-level pure-Python API for [de]serializing a billion different mesh formats. Who doesn't want that?
  * `dev-python/scooby`. Scooby is kinda useless for basically everyone, but the [PyVista lead developer wrote it](https://banesullivan.com) so PyVista currently requires it. *What can you do.* 
  * `dev-python/transforms3d`, with full unit test support. Transforms3d is [currently unmaintained](https://github.com/pyvista/pyvista/issues/1388). That's not great, but [upstream PyVista developers are aware and actively removing transforms3d as a dependency](https://github.com/pyvista/pyvista/issues/1388#issuecomment-863727742). For now, it's mostly fine. When they remove it, so will we!

The PyVista and Scooby ebuilds currently lack `test` support, which isn't great. Unfortunately, adding `test` support for either would require packaging even *more* Python packages, which also isn't great. In the end, I opted to simplify @AndrewAmmerlaan's life by temporarily omitting `test` support from both. When someone else packages their test dependencies, we can happily revisit this.

Thanks for all the ongoing efforts, everyone! You're all amazing. :heart_eyes:

### CODE REVIEW

**Comment by leycec** [comment]
3f7af47 trivially patches the Scooby description, which I'd initially forgotten. This should now be ready for a ~~violent public beatdown~~ friendly code review.

File: dev-python/transforms3d/transforms3d-0.3.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{8..10} )
```
Comment (AndrewAmmerlaan) [inline dev-python/transforms3d/transforms3d-0.3.1.ebuild:6]:
```
* Log file: /var/log/portage/build/dev-python/transforms3d-0.3.1:20210618-082918.log

* DISTUTILS_USE_SETUPTOOLS value is probably incorrect
*   have:     DISTUTILS_USE_SETUPTOOLS=bdepend (or unset)
*   expected: DISTUTILS_USE_SETUPTOOLS=no
```

**Comment by AndrewAmmerlaan** [comment]
Thanks, other than the DISTUTILS_USE_SETUPTOOLS thing this looks fine :+1: 

Next time could you please put each package in a separate commit

**Comment by akaszynski** [comment]
Hello, `pyvista` maintainer/author here.  A few notes regarding this PR:

> This commit adds PyVista, popular quasi-official Python bindings for VTK, a popular C++ framework for scientific 3D visualization. Both are maintained by Kitware – an academic open-source company sorta resembling the Qt Company in structure and practice.

I *wish* `pyvista` was maintained by Kitware.  As it stands, we're still an unofficial package.  Ideally, one day Kitware will extend their blessing on our project, especially as the other co-author of `pyvista`, @banesullivan, now works there.  @Kitware is super busy with maintaining VTK and working on a variety of projects, and thus `pyvista` as born out of the necessity of making VTK more accessible to the scientific visualization community.

> Scooby is kinda useless for basically everyone
I got a good laugh out of that.  I'll admit its adoption is fairly limited, but it has been quite helpful with generating reports and is used in a variety of projects to make issue reports easier.  This could be made optional if necessary to get into gentoo (which we're quite honored to be part of!).

**Comment by leycec** [comment]
> Ideally, one day Kitware will extend their blessing on our project, especially as the other co-author of pyvista, @banesullivan, now works there.

*I knew this was too good to be true.* I saw that Kitware adopted Bane (...the open-source superhero name I wish I had); I saw that Bane invented PyVista; I made bad assumptions from sound premises.

PyVista is the natural next step for VTK. Have no doubt that Kitware will eventually fold PyVista into their portfolio! PyVista is the sanest road forward for Kitware to stay competitive on web dashboards, which is the natural endgame for science in Python. You've got this one in the bag. :handbag: 

Actually, while we're on the pertinent subject: if you could chime in on [this Panel feature request for bidirectional VTK synchronization](https://github.com/holoviz/panel/issues/2363), my wife and I would be *incredibly* obliged. It'd be great if Panel and PyVista were integrated at a somewhat deeper level – enabling client-side JavaScript events like interactively picked cells to percolate down into server-side Python callbacks. A techbro can dream, right?

> I got a good laugh out of that.

Has anyone told you that you're positive, forthright, and awesome, @akaszynski?

It's a perennial joy conversing with you, because you always embrace the open-hearted position rather than assuming a defensive or hostile posture. What I'm saying is... you're basically online Aikido.

> This could be made optional if necessary to get into gentoo (which we're quite honored to be part of!).

Oh, you. Flattery will get you everywhere! Please don't sweat the small stuff. Scooby was trivial to package and will pay dividends elsewhere once other downstream packages begin requiring it. *Which they'd better.*

<sup>We now return to our regularly scheduled Gentoo PR.</sup>

File: dev-python/transforms3d/transforms3d-0.3.1.ebuild (hunk 1 above)
Comment (leycec) [reply dev-python/transforms3d/transforms3d-0.3.1.ebuild:6]: **Woops.** This is me mopping sweat from my forehead. That's obviously wrong, so I'll get that patched right up. Thanks for the last-minute save... *as always.* :+1:

**Comment by leycec** [comment]
> Thanks, other than the `DISTUTILS_USE_SETUPTOOLS` thing this looks fine :+1: 

Thank *you* for the logfile catch. I'm so disappointed I missed that. :relieved: 

> Next time could you please put each package in a separate commit

**Definitely!** I *knew* I was forgetting something there. To preserve everyone's sanity, I'll do that right now and split this out into package-sized commits.

**Comment by leycec** [comment]
**Bam!** `transforms3d-0.3.1.ebuild` now declares the expected `DISTUTILS_USE_SETUPTOOLS=no`. Moreover, individual packages have now been split into their own commits as required by sanity.

Thanks for the prompt code review, @AndrewAmmerlaan, and the phenomenal Python bindings, @akaszynski. :smile:

**Comment by AndrewAmmerlaan** [comment]
> Thanks for the prompt code review, @AndrewAmmerlaan, and the phenomenal Python bindings, @akaszynski. smile

Thank you, for another excellent Pull Request :+1:

**Pull request closed without merging**